src/p2p/swarm.rs

```diff
@@ -77,15 +97,15 @@ impl SwarmApi {
 
     pub fn disconnect(&mut self, address: Multiaddr) -> Option<Disconnector> {
         log::trace!("disconnect {}", address);
-        self.connections.remove(&address);
-        let peer_id = self
-            .connections
-            .get(&address)
-            .map(|conn| conn.peer_id.clone());
-        if let Some(peer_id) = &peer_id {
-            self.connected_peers.remove(peer_id);
```

Comment:
well, removing this seems to have broken everything up.

Reply:
yes, turns out the `NetworkBehaviour::inject_disconnected` and `NetworkBehaviour::inject_connection_closed` do *not* get called when using the ban/unban hack to boot a peer but a `SwarmEvent` is returned polling the swarm for each of the connections..

Reply:
Created an issue at rust-libp2p, resolving